### PR TITLE
feat: TOOLS-2516 add check for stop writes at startup

### PR DESCRIPTION
**Pull request by jdogmcsteezy**
In the AerospikeShell init I no longer call logger.critical. Instead I set self.connected = False and let the caller handle the exit code. Other than that most the changes are related to type annotations

### CODE REVIEW

File: asadm.py [hunk 1]
```diff
@@ -138,7 +140,8 @@ async def __init__(
                         "You have not specified any collectinfo path. Usage: asadm -c -f <collectinfopath>"
                     )
                     await self.do_exit("")
```
Comment (jdogmcsteezy) [inline asadm.py:142]: make sure connected is used in -cf mode

File: asadm.py [hunk 2]
```diff
@@ -194,25 +196,41 @@ async def __init__(
...
-                        self.intro += (
-                            terminal.fg_red()
-                            + "Extra nodes in alumni list: %s"
+                        logger.warning(
+                            terminal.fg_yellow()
+                            + "Some nodes have become unreachable by other nodes in the cluster. Check their peers lists: %s"
                             % (", ".join(cluster_down_nodes))
                             + terminal.fg_clear()
-                            + "\n"
                         )
```
Comment (jdogmcsteezy) [inline asadm.py:215]: Check on coroutine RuntimeWarning un-awaited

**Review by jdogmcsteezy — COMMENTED** [review]
Some changes needed after first review with @dwelch-spike

**Review by dwelch-spike — APPROVED** [review]
LGTM

**Comment by dwelch-spike** [comment]
holding for tools 8.6

**Pull request merged**